src/util.c

```diff
@@ -40,17 +40,31 @@
  */
 char *xmpp_strdup(const xmpp_ctx_t *ctx, const char *s)
 {
-    size_t len;
+    return xmpp_strndup(ctx, s, strlen(s));
+}
+
+/** Duplicate a string with a maximum length.
+ *  This function replaces the standard strndup library call with a version
+ *  that uses the Strophe context object's allocator.
+ *
+ *  @param ctx a Strophe context object
+ *  @param s a string
+ *  @param n the maximum length of the string to copy
+ *
+ *  @return a new allocates string with the same data as s or NULL on error
+ */
+char *xmpp_strndup(const xmpp_ctx_t *ctx, const char *s, size_t len)
+{
     char *copy;
 
-    len = strlen(s);
     copy = xmpp_alloc(ctx, len + 1);
```

Comment:
strndup()  must handle strings that are shorter than `len`. This can be done in this way for example:
```
n = strlen(s);
if (n < len)
    len = n;
...
```
And xmpp_strdup() can be implemented like this:
```
{
    xmpp_strndup(ctx, s, SIZE_MAX);
}
```

Reply:
thx! done